### PR TITLE
Error logging

### PR DESCRIPTION
**Pull request by yuth**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/package.json [hunk 1]
```diff
@@ -17,63 +17,49 @@
   ],
   "private": true,
   "scripts": {
-    "e2e": "npm run setup-profile && jest",
+    "e2e": "npm run setup-profile && jest --verbose",
```
Comment (attilah) [inline packages/amplify-e2e-tests/package.json]: Why is verbose needed?

File: packages/amplify-e2e-tests/package.json [hunk 2]
```diff
@@ -17,63 +17,49 @@
...
     "setup-profile": "ts-node ./src/configure_tests.ts"
   },
   "dependencies": {
+    "@types/jest": "^25.1.3",
```
Comment (attilah) [inline packages/amplify-e2e-tests/package.json]: It is not needed based on @SwaySway PR what we just merged and should be in devdeps if would be needed.

File: packages/amplify-e2e-tests/src/asciinema-recorder.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+};
+export type RecordingFrame = [number, 'o' | 'i', string];
+export type Recording = {
+  header: RecordingHeader;
+  frames: RecordingFrame[];
+};
+
+export class Recorder {
+  private isPaused: boolean = false;
+  private proc: pty.IPty;
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/asciinema-recorder.ts]: 'childProcess'

File: packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-setup.ts [hunk 4]
```diff
@@ -57,14 +54,13 @@ function amplifyAppAngular(projRoot: string, verbose: Boolean = isCI() ? false :
...
   return new Promise((resolve, reject) => {
-    nexpect
-      .spawn(spawnCommand, { cwd: projRoot, stripColors: true, verbose })
+    spawn(spawnCommand, [], { cwd: projRoot, stripColors: true, verbose })
       .wait('What type of app are you building')
-      .sendline('\r')
+      .sendLine('')
       .wait('What javascript framework are you using')
-      .sendline('jjj\r')
+      .sendLine('jjj')
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-setup.ts]: replace with ARROW constant?

File: packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-setup.ts [hunk 5]
```diff
@@ -57,14 +54,13 @@ function amplifyAppAngular(projRoot: string, verbose: Boolean = isCI() ? false :
...
 
-function amplifyAppReact(projRoot: string, verbose: Boolean = isCI() ? false : true) {
+function amplifyAppReact(projRoot: string, verbose: boolean = !isCI()) {
   return new Promise((resolve, reject) => {
-    nexpect
-      .spawn(spawnCommand, { cwd: projRoot, stripColors: true, verbose })
+    spawn(spawnCommand, [], { cwd: projRoot, stripColors: true, verbose })
       .wait('What type of app are you building')
-      .sendline('\r')
+      .sendLine('')
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-setup.ts]: sendCarriageReturn() could make it more expressive when writing tests and also then sendLine could make the only argument as required and empty passed in argument could be treated as invalid.

File: packages/amplify-e2e-tests/src/categories/api.ts [hunk 6]
```diff
@@ -82,32 +80,31 @@ export function addApiWithSchema(cwd: string, schemaFile: string, verbose: boole
...
+      .sendLine('')
       .wait(/.*Enter a description for the API key.*/)
-      .sendline('\r')
+      .sendLine('')
       .wait(/.*After how many days from now the API key should expire.*/)
-      .sendline('\r')
+      .sendLine('')
       .wait(/.*Do you want to configure advanced settings for the GraphQL API.*/)
-      .sendline('\x1b[B') // Down
+      .sendLine('\x1b[B') // Down
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/categories/api.ts]: ARROW constant?

File: packages/amplify-e2e-tests/src/categories/api.ts [hunk 7]
```diff
@@ -127,42 +124,41 @@ export function updateApiSchema(cwd: string, projectName: string, schemaName: st
...
+      .sendLine('')
       .wait(/.*Enter a description for the API key.*/)
-      .sendline('description')
+      .sendLine('description')
       .wait(/.*After how many days from now the API key should expire.*/)
-      .sendline('300')
+      .sendLine('300')
       .wait(/.*Do you want to configure advanced settings for the GraphQL API.*/)
-      .sendline('\x1b[B') // Down
+      .sendLine('\x1b[B') // Down
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/categories/api.ts]: ARROW constant?

File: packages/amplify-e2e-tests/src/categories/api.ts [hunk 8]
```diff
@@ -177,26 +173,25 @@ export function updateApiWithMultiAuth(cwd: string, settings: any, verbose: bool
...
+      .sendLine('')
       .wait(/.*Enter a description for the API key.*/)
-      .sendline('\r')
+      .sendLine('')
       .wait(/.*After how many days from now the API key should expire.*/)
-      .sendline('\r')
+      .sendLine('')
       .wait(/.*Do you want to configure advanced settings for the GraphQL API.*/)
-      .sendline('\x1b[B') // Down
+      .sendLine('\x1b[B') // Down
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/categories/api.ts]: ARROW constant?

File: packages/amplify-e2e-tests/src/categories/api.ts [hunk 9]
```diff
@@ -177,26 +173,25 @@ export function updateApiWithMultiAuth(cwd: string, settings: any, verbose: bool
...
       .wait(/.*Configure additional auth types.*/)
-      .sendline('n')
+      .sendLine('n')
       .wait(/.*Configure conflict detection.*/)
-      .sendline('y')
+      .sendLine('y')
       .wait(/.*Select the default resolution strategy.*/)
-      .sendline('\x1b[B') // Down
+      .sendLine('\x1b[B') // Down
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/categories/api.ts]: ARROW constant?

File: packages/amplify-e2e-tests/src/categories/function.ts [hunk 10]
```diff
@@ -103,38 +102,38 @@ function _coreFunction(cwd: string, settings: any, action: 'create' | 'update',
...
+  return chain.sendLine('');
 }
 
-function addLambdaTrigger(chain: nexpect.IChain, cwd: string, settings: any, verbose: boolean) {
+function addLambdaTrigger(chain: ExecutionContext, cwd: string, settings: any) {
   const res = chain
     .wait('What event source do you want to associate with Lambda trigger')
     // Amazon DynamoDB Stream
-    .sendline(settings.triggerType == 'Kinesis' ? '\x1b[B\r' : '\r')
+    .sendLine(settings.triggerType == 'Kinesis' ? '\x1b[B\r' : '')
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/categories/function.ts]: ===

File: packages/amplify-e2e-tests/src/cli-test-environment.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
+// my-custom-environment
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/cli-test-environment.js]: ?

File: packages/amplify-e2e-tests/src/codegen/add.ts [hunk 12]
```diff
@@ -1,25 +1,25 @@
...
-      run.wait('Choose the code generation language target').sendline('');
+      run.wait('Choose the code generation language target').sendLine('');
     }
     run
       .wait('Enter the file name pattern of graphql queries, mutations and subscriptions')
-      .sendline('\r')
+      .sendLine('')
       .wait('Do you want to generate/update all possible GraphQL operations')
-      .sendline('Yes')
+      .sendLine('Yes')
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/codegen/add.ts]: 'y'

File: packages/amplify-e2e-tests/src/nexpect-reporter.js [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+      if (testBlock.hooks.afterEach) {
+        after = getLogs(testBlock.hooks.afterEach).map(afe => ({ ...afe, name: `${prefixStr} --> AfterEach` }));
+      }
+    }
+  }
+
+  return [...before, ...children, ...after];
+};
+
+class MyCustomReporter {
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/nexpect-reporter.js]: ?

File: packages/amplify-e2e-tests/package.json (hunk 1 above)
Comment (yuth) [reply packages/amplify-e2e-tests/package.json]: Makes the circle ci logs better

File: packages/amplify-e2e-tests/src/__tests__/storage.test.ts [hunk 14]
```diff
@@ -60,16 +60,16 @@ describe('amplify add/update storage(DDB) with GSI', () => {
...
+    projRoot = await createNewProjectDir('ddb-add-update');
   });
 
   afterEach(async () => {
     await deleteProject(projRoot);
     deleteProjectDir(projRoot);
   });
 
-  it('init a project and add/update ddb table with & without trigger', async () => {
+  it.only('init a project and add/update ddb table with & without trigger', async () => {
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/__tests__/storage.test.ts]: .only leftover
